src/halmos/config.py

```diff
@@ -549,7 +549,17 @@ def parse_dict(self, parsed: dict, source: str = "halmos.toml") -> dict:
                 )
                 sys.exit(2)
 
-        return {k.replace("-", "_"): v for k, v in data.items()}
+        # gather custom actions
+        actions = {
+            field.name: field.metadata.get("action", None) for field in fields(Config)
```

Comment:
instead of having a mostly `None` dict, could filter entries:

```suggestion
            field.name: field.metadata["action"] for field in fields(Config) if "action" in field.metadata
```

that way your actions dict contain only the fields with a custom action